core/src/main/java/bio/terra/pearl/core/model/survey/StudyEnvironmentSurvey.java

```diff
@@ -33,4 +35,77 @@ public Versioned versionedEntity() {
     public void updateVersionedEntityId(UUID surveyId) {
         setSurveyId(surveyId);
     }
+
+
+    @Override
+    public String getStableId() {
+        if (survey == null) {
+            return null;
+        }
+        return survey.getStableId();
+    }
+
+    @Override
+    public Integer getVersion() {
+        if (survey == null) {
+            return null;
+        }
+        return survey.getVersion();
+    }
+
+    @Override
+    public RecurrenceType getRecurrenceType() {
+        if (survey == null) {
+            return null;
```

Comment:
I'm not super happy with this - I think it's awkward to have all the null checks. But, at the same time, I think most tasks wouldn't be portal attached like surveys are, so I think it makes most sense to treat everything as a flat TaskConfig with surveys being the odd one out.